app/src/org/commcare/CommCareApplication.java

```diff
@@ -234,6 +238,12 @@ public void onCreate() {
         LocalePreferences.saveDeviceLocale(Locale.getDefault());
     }
 
+    private void initTls12IfNeeded() {
+        if (Build.VERSION.SDK_INT >= 16 && Build.VERSION.SDK_INT < 22) {

```

Comment:
Should not it be [less than 20](https://developer.android.com/reference/javax/net/ssl/SSLSocket.html#protocols) ?

Reply:
The Stripe team that I got the code from identified that there were some 20/21 API devices which still didn't have the update. I think from the data that our audience of devices which are off KitKat but not on a significantly newer version is pretty small, so I figured I'd take their word for it. 

I should add links to the sources, though, which will clarify where that info came from, thanks for catching.